core/src/blocktree.rs

```diff
@@ -459,12 +583,57 @@ impl Blocktree {
         Ok(())
     }
 
+    fn insert_coding_shred(
+        &self,
+        set_index: u64,
+        num_data: usize,
+        num_coding: usize,
+        prev_inserted_coding_shreds: &mut HashMap<(u64, u64), Shred>,
+        erasure_metas: &mut HashMap<(u64, u64), ErasureMeta>,
+        index_meta: &mut Index,
+        shred: Shred,
+        write_batch: &mut WriteBatch,
+    ) {
+        let slot = shred.slot();
+        let shred_index = u64::from(shred.index());
+
+        let erasure_config = ErasureConfig::new(num_data, num_coding);
+
+        let erasure_meta = erasure_metas.entry((slot, set_index)).or_insert_with(|| {
```

Comment:
Is the `set_index` guaranteed to be unique across FEC blocks within the same slot?